n2y/main.py

```diff
@@ -1,36 +1,124 @@
 import os
 import sys
 import argparse
+import re
 
 import yaml
+import pandoc
 
-from n2y import notion, simplify
+from n2y import converter, notion, simplify
 
 
 def main():
+    parser = argparse.ArgumentParser(description="Move data from Notion into YAML/markdown",
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("database", help="The Notion database id or share url")
+    parser.add_argument("--output", '-o',
+                        choices=["yaml", "markdown"], default="yaml",
+                        help=("Select output type\n"
+                              "  yaml - print yaml to stdout\n"
+                              "  markdown - create a markdown file per page"))
+    parser.add_argument("--image-path", help="Specify path where to save images")
+    parser.add_argument("--image-web-path", help="Web path for images")
+    parser.add_argument("--plugins", help="Plugin file")
+    parser.add_argument("--name-column", default='title',
+                        help=("Name of column containing the page name."
+                              "Lowercase letter and numbers. Replace spaces with underscore."))
+    args = parser.parse_args()
+
     ACCESS_TOKEN = os.environ.get("NOTION_ACCESS_TOKEN", None)
     if ACCESS_TOKEN is None:
         print("No NOTION_ACCESS_TOKEN environment variable is set", file=sys.stderr)
         return 1
 
-    parser = argparse.ArgumentParser(description="Move data from Notion into YAML files")
-    parser.add_argument("database", help="The Notion database id or share url")
-    parser.add_argument(
-        "--raw", action='store_true',
-        help="Dump the raw notion API data",
-    )
-    args = parser.parse_args()
     database_id = notion.id_from_share_link(args.database)
+
+    converter.IMAGE_PATH = args.image_path
+    converter.IMAGE_WEB_PATH = args.image_web_path
+    if args.plugins:
+        converter.load_plugins(args.plugins)
+
     client = notion.Client(ACCESS_TOKEN)
+
     raw_rows = client.get_database(database_id)
-    if args.raw:
-        print(yaml.dump(raw_rows))
-        return 0
+    first_row_flattened = simplify.flatten_database_row(raw_rows[0])
+
+    def available_columns():
```

Comment:
Why use a function for this instead of just calculating the value? Is this to avoid calculating the value if it doesn't end up being used?

Reply:
Yes, there are two places where the available columns may be shown, but there is usually no need to generate this list. Also, I imagine this function will grow in future versions as we would want deeper checks.